### PR TITLE
16 implement user registration

### PR DESCRIPTION
**Pull request by CJStuart**

### CODE REVIEW

File: src/api/util.ts [hunk 1]
```diff
@@ -14,44 +14,51 @@ export class StatusError extends Error {
...
 
-    async addHeaders(headers: Headers) {
-        this.headers = new Headers({...this.headers});
+    async addHeaders(headers: Record<string, string>) {
+        this.headers = {...this.headers, ...headers};
     }
 
     async setToken(token: string) {
-        this.setHeader('Authorization', `Token ${token}`);
+        this.setHeader("Authorization", `Token ${token}`);
```
Comment (CJStuart) [inline src/api/util.ts:34]: This is not going to work until pokeclicker/backend#51 is fixed..

**Review by SanderMeister — APPROVED** [review]

File: src/api/util.ts [hunk 2]
```diff
@@ -65,3 +72,19 @@ export function url(parts: TemplateStringsArray, ...params: any[]): string {
...
+ */
+function mergeInit(a?: RequestInit, b?: RequestInit): RequestInit {
+    const headers = {};
+    if(a != null) {
+        Object.assign(headers, a.headers);
+    }
+    if(b != null) {
+        Object.assign(headers, b.headers);
+    }
+    return {...a, ...b, headers};
```
Comment (SimonBaars) [inline src/api/util.ts:89]: What does the ... do?

**Review by SimonBaars — APPROVED** [review]

**Comment by SimonBaars** [comment]
closes #16

File: src/api/util.ts (hunk 2 above)
Comment (CJStuart) [reply src/api/util.ts:89]: `{...a}` creates a new object, and copies over all properties from `a` to the new object. 

It resolves in left to right order, so in the case of `{...a, ...b, headers}` the values of `b` override the values of `a`, and finally `headers` is set.

**Pull request merged**